ext/zip/php_zip.c

```diff
@@ -2293,8 +2312,8 @@ static ZIPARCHIVE_METHOD(getCommentIndex)
 	}
 
 	PHP_ZIP_STAT_INDEX(intern, index, 0, sb);
-	comment = zip_get_file_comment(intern, index, &comment_len, (int)flags);
-	RETURN_STRINGL((char *)comment, (zend_long)comment_len);
+	comment = zip_file_get_comment(intern, index, &comment_len, (zip_flags_t)flags);
+	RETURN_STRINGL((char *)comment, MIN(comment_len, ZEND_LONG_MAX));
```

Comment:
This `MIN` looks unnecessary?

Reply:
`comment_len` is `uint32_t`, so could overflow `zend_long` on 32bit architectures. This `MIN()` is supposed to prevent that (and to silently truncate instead).

Reply:
The string length is size_t though, not zend_long. On platforms we support we should always have SIZE_MAX >= UINT32_MAX.

Reply:
Ah, of course! Guess I have been confused by the (zend_long) cast in the old code.